compose/service.py

```diff
@@ -425,27 +427,32 @@ def _containers_have_diverged(self, containers):
 
         return has_diverged
 
-    def _execute_convergence_create(self, scale, detached, start, project_services=None):
-            i = self._next_container_number()
+    def _execute_convergence_create(self, scale, detached, start):
 
-            def create_and_start(service, n):
-                container = service.create_container(number=n, quiet=True)
-                if not detached:
-                    container.attach_log_stream()
-                if start:
-                    self.start_container(container)
-                return container
+        i = self._next_container_number()
 
-            containers, errors = parallel_execute(
-                [ServiceName(self.project, self.name, index) for index in range(i, i + scale)],
-                lambda service_name: create_and_start(self, service_name.number),
-                lambda service_name: self.get_container_name(service_name.service, service_name.number),
-                "Creating"
-            )
-            for error in errors.values():
-                raise OperationFailedError(error)
+        def create_and_start(service, n):
+            container = service.create_container(number=n, quiet=True)
+            if not detached:
+                container.attach_log_stream()
+            if start:
+                self.start_container(container)
+            return container
 
-            return containers
+        containers, errors = parallel_execute(
+            [ServiceName(self.project, self.name, number, generate_random_id()) for number in [
```

Comment:
Maybe I'm missing something but why use `number` here?
```
- [ServiceName(self.project, self.name, number, generate_random_id()) for number in [index for index in range(i, i + scale)]]
+ [ServiceName(self.project, self.name, index, generate_random_id()) for index in range(i, i + scale)]
```

Reply:
You're not missing anything, I screwed up. Thanks!